src/Security/Authentication/WsFederation/src/WsFederationHandler.cs

```diff
@@ -98,7 +98,7 @@ protected override async Task HandleChallengeAsync(AuthenticationProperties prop
         var wsFederationMessage = new WsFederationMessage()
         {
             IssuerAddress = _configuration.TokenEndpoint ?? string.Empty,
-            Wtrealm = Options.Wtrealm,
+            Wtrealm = Options.Wtrealm!,
```

Comment:
Is this the right fix? It seems wrong.

Reply:
It looks like the change that broke this is

https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/commit/c058737642f6e04aab80fef5f15de118e7bb2adc

@brentschmaltz @westin-m @FuPingFranco - are you certain that `WsFederationMessage.Wtrealm` needs to be non-nullable?

Reply:
@eerhardt When requesting a SecurityToken, wtrealm must contain a uri.
http://[docs.oasis-open.org/wsfed/federation/v1.2/os/ws-federation-1.2-spec-os.html#_Toc223175023](http://docs.oasis-open.org/wsfed/federation/v1.2/os/ws-federation-1.2-spec-os.html#_Toc223175023)

Reply:
@halter73 @Tratcher - any thoughts here? Why is `Options.Wtrealm` annotated as `string?` when `WsFederationMessage.Wtrealm` is a non-nullable `string`. Do we have our nullable annotations incorrect in ASP.NET?

Reply:
@eerhardt will you fix it on your end?

Reply:
I've opened https://github.com/dotnet/aspnetcore/issues/50242 to track annotating `Wtrealm` correctly, and added links to those 2 locations in the code. I believe the `Wctx` below is a bug in IdentityModel.